src/analysis/complex/schwarz.lean

```diff
@@ -0,0 +1,167 @@
+/-
+Copyright (c) 2022 Yury G. Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury G. Kudryashov
+-/
+import analysis.complex.abs_max
+import analysis.complex.removable_singularity
+
+/-!
+# Schwarz lemma
+
+In this file we prove several versions of the Schwarz lemma.
+
+* `complex.norm_deriv_le_div_of_maps_to_ball`, `complex.abs_deriv_le_div_of_maps_to_ball`: if
+  `f : ℂ → E` sends an open disk with center `c` and a positive radius `R₁` to an open ball with
+  center `f c` and radius `R₂`, then the absolute value of the derivative of `f` at `c` is at most
+  the ratio `R₂ / R₁`;
+
+* `complex.dist_le_div_mul_dist_of_maps_to_ball`: if `f : ℂ → E` sends an open disk with center `c`
+  and radius `R₁` to an open disk with center `f c` and radius `R₂`, then for any `z` in the former
+  disk we have ``dist (f z) (f c) ≤ (R₂ / R₁) * dist z c`;
+
+* `complex.abs_deriv_le_one_of_maps_to_ball`: if `f : ℂ → ℂ` sends an open disk of positive radius
+  to itself and the center of this disk to itself, then the absolute value of the derivative of `f`
+  at the center of this disk is at most `1`;
+
+* `complex.dist_le_dist_of_maps_to_ball`: if `f : ℂ → ℂ` sends an open disk to itself and the center
+  `c` of this disk to itself, then for any point `z` of this disk we have `dist (f z) c ≤ dist z c`;
+
+* `complex.abs_le_abs_of_maps_to_ball`: if `f : ℂ → ℂ` sends an open disk with center `0` to itself,
+  the for any point `z` of this disk we have `abs (f z) ≤ abs z`.
+
+## Implementation notes
+
+We prove some versions of the Schwarz lemma for a map `f : ℂ → E` taking values in any normed space
+over complex numbers.
+
+## TODO
+
+* Prove that these inequalities are strict unless `f` is an affine map.
+
+* Prove that any diffeomorphism of the unit disk to itself is a Möbius map.
+
+## Tags
+
+Schwarz lemma
+-/
+
+open metric set function filter topological_space
+open_locale topological_space
+
+namespace complex
+
+section space
+
+variables {E : Type*} [normed_group E] [normed_space ℂ E] {R R₁ R₂ : ℝ} {f : ℂ → E} {c z : ℂ}
+
+/-- An auxiliary lemma. -/
```

Comment:
I don't think this docstring is helpful right now. Maybe it can contain a pointer to the decls that it is an aux_lemma for? But I would also be fine with simply dropping it.

Reply:
```suggestion
/-- An auxiliary lemma for `complex.norm_dslope_le_div_of_maps_to_ball`. -/
```